api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -111,17 +111,87 @@ public CohortDefinition getPatientsWhoInitiatedPreTarvAtAfacilityDuringCurrentMo
     cd.addSearch(
         "A2I",
         map(sqlCohortDefinition, "startDate=${startDate},endDate=${endDate},location=${location}"));
-    cd.addSearch("A2III", map(getPatientsWithTransferFromOtherHF(), "locationList=${location}"));
     cd.addSearch(
         "A2II",
         map(
-            getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreArtStartDate(),
-            "endDate=${endDate},location=${location}"));
+            getPatientsEnrolledInPreArtOrRegisteredInClinicalProcessOpening(),
```

Comment:
From the spec:
> ◦ Select all patients registered in encounter “Master Card – Ficha Resumo” (encounter id 53) who have the following conditions: 
> ▪ “Pre-ART Start Date” (PT”: “Data do Inicio Pre-TARV”) (Concept ID 23808)
> ▪ enrolled in PRE-ART Program (program_id=1, from patient program table) with enrollement date (date_enrolled) OR
> ▪ registered in “Abertura de Processo Clínico” (encounter type id 5 or 7) with “Data de Registo” (encounter_datetime) 
> ▪ **with the most earliest date from above criterias in the reporting period (>=startDate and <=endDate)**

Implementation is missing the part highlighted in bold. 

You will probably need to do a union of `A2I` and `A2II` in a single query, check which of the criterias has the minimum date then check if this date falls between the reporting period.

You can use this part of a query as reference:
https://github.com/esaude/openmrs-module-eptsreports/blob/019197fc89034447ec70eb041796bf0903757f1d/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxMlQueries.java#L250-L295

This one looks for the most recent date between 3 criterias and adds 28 days to the date found.